R/fit_uniform.R

```diff
@@ -0,0 +1,128 @@
+#' Fit a uniform distribution to a histogram
+#'
+#' @param x numeric vector representing the density of a histogram
+#' @param metric one of `mle`, `jaccard`, `intersection`, `ks`, `mse`, `chisq`
+#'
+#' @return a list with the following data
+#' \describe{
+#'     \item{par}{a character string denoting the region_id of the Histogram}
+#'     \item{dist}{the distribution name}
+#'     \item{metric}{the metric used to fit the distribution}
+#'     \item{value}{the fitted value of the metric function}
+#'     \item{dens}{a function that returns the density of the fitted distribution}
+#' }
+#' @export fit_uniform
+fit_uniform <- function(x, metric=c('mle', 'jaccard', 'intersection', 'ks', 'mse', 'chisq')) {
+  UseMethod('fit_uniform')
+}
+
+
+#' @exportS3Method fit_uniform numeric
+fit_uniform.numeric <- function(x, metric=c('mle', 'jaccard', 'intersection', 'ks', 'mse', 'chisq')) {
+
+  interval_midpoint <- seq(1, length(x), 1)
+  fit_uniform_helper(
+    x = x,
+    interval_start = interval_midpoint - 0.5,
+    interval_end = interval_midpoint + 0.5,
+    interval_midpoint = interval_midpoint,
+    metric = metric
+  )
+
+}
+
+#' @exportS3Method  fit_uniform table
+fit_uniform.table <- fit_uniform.numeric
+
+#' @exportS3Method fit_uniform Histogram
```

Comment:
I think it makes sense to use the `@rdname` roxygen tag so that the documentation pages are linked. For example, see: https://github.com/uclahs-cds/public-R-HistogramZoo/blob/main/R/moments.R#L7 Check out what the documentation pages look like if you do `weighted.sd`. I would use `@rdname fit_uniform` for all of these methods (numeric, Histogram, GenomicHistogram, etc.)